charts/k8ssandra/values.yaml

```diff
@@ -468,6 +468,8 @@ reaper:
   # will be configured in Cassandra with credentials only created for Reaper
   # in order to limit access.
   enabled: true
+  # -- The name of the service account to use for Reaper pods.
+  serviceAccount: default
```

Comment:
serviceAccount is commented out for cleaner and client and is set to empty string for c*.  Double checking that you want Reaper to have the default value or should this be commented out as well?

Reply:
Good observations. To make things a little more consistent, I can default them all to the empty string. In the case of Reaper, Stargate, and Cassandra, this will result in using the `default` service account. In the case of the cleaner and client this would result in the creation of their respective service accounts. Unfortunately there are some inconsistencies that would be nice to address, but I think that would need to be post 1.3.0.

There is another problem with the cleaner and client service accounts. I explained in the description why I didn't use the `k8ssandra-common.serviceAccount` template. If the user specifies a different service account I need to update the RoleBinding template to use the specified ones.

Reply:
I thought about this some more, and here things I think we need to do post 1.3.0:

* Refactor the `k8ssandra-common.serviceAccount` template so it doesn't assume a scope of `.Values`. This will make it more flexible and easier to use elsewhere.
    * Service accounts created with the `k8ssandra-common.serviceAccount` can have both the service account name and image pull secrets configured.
* Create a service account for Cassandra pods using the `k8ssandra-common.serviceAccount` template
    * Allow the user to specify a different service account name to use
* Create a service account for Stargate pods using the `k8ssandra-common.serviceAccount` template
    * Allow the user to specify a different service account name to use
* Create a service account for Reaper pod using the `k8ssandra-common.serviceAccount` template
    * Allow the user to specify a different service account name to use
* Create a service account for cleaner pod using the `k8ssandra-common.serviceAccount` template
    * Allow the user to specify a different service account name to use 
* Create a service account for crd upgrader (i.e., client) pod using the `k8ssandra-common.serviceAccount` template
    * Allow the user to specify a different service account name to use 

In summary, we should be creating a service account for every pod that we are deploying and it should be done using the `k8ssandra-common.serviceAccount` template.

@jdonenine I definitely think we need to get these ticketed. I don't know how much bandwidth we have to address them though with everything else that is in flight.

Reply:
@jsanda got it... I'll take what you laid out here and get it into some issues to consider post-1.3